disnake/automod.py

```diff
@@ -127,11 +135,22 @@ class AutoModBlockMessageAction(AutoModAction):
 
     _metadata: AutoModBlockMessageActionMetadata
 
-    def __init__(self) -> None:
+    def __init__(self, custom_message: Optional[str] = None) -> None:
         super().__init__(type=AutoModActionType.block_message)
 
+        if custom_message is not None:
+            self._metadata["custom_message"] = custom_message
+
+    @property
+    def custom_message(self) -> str:
+        """:class:`str`: The custom message to send to the user when the rule is triggered.
+
+        .. versionadded:: 2.9
+        """
+        return self._metadata.get("custom_message", "")
```

Comment:
```suggestion
    @property
    def custom_message(self) -> Optional[str]:
        """Optional[:class:`str`]: The custom message to show to the user when the rule is triggered.

        .. versionadded:: 2.9
        """
        return self._metadata.get("custom_message")
```
Better to make this optional, for clarity